SPIFlashLogger.device.lib.nut

```diff
@@ -0,0 +1,847 @@
+// MIT License
+//
+// Copyright 2017 Electric Imp
+//
+// SPDX-License-Identifier: MIT
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+// Using `const`s instead of `static`s for performance
+const SPIFLASHLOGGER_SECTOR_SIZE = 4096; // Size of sectors
+const SPIFLASHLOGGER_SECTOR_METADATA_SIZE = 3; // Size of metadata at start of sectors
+const SPIFLASHLOGGER_SECTOR_BODY_SIZE = 4093; // Size of writeable memory / sector
+const SPIFLASHLOGGER_CHUNK_SIZE = 256; // Number of bytes we write / operation
+
+const SPIFLASHLOGGER_OBJECT_MARKER = "\x00\xAA\xCC\x55";
+const SPIFLASHLOGGER_OBJECT_MARKER_SIZE = 4;
+
+const SPIFLASHLOGGER_OBJECT_HDR_SIZE = 7; // SPIFLASHLOGGER_OBJECT_MARKER (4 bytes) + size (2 bytes) + crc (1 byte)
+const SPIFLASHLOGGER_OBJECT_MIN_SIZE = 6; // SPIFLASHLOGGER_OBJECT_MARKER (4 bytes) + size (2 bytes)
+
+const SPIFLASHLOGGER_SECTOR_DIRTY = 0x00; // Flag for dirty sectors
+const SPIFLASHLOGGER_SECTOR_CLEAN = 0xFF; // Flag for clean sectors
```

Comment:
The constant has never been used. Why?

Reply:
Thanks, I'll drop them, it is from old implementation